File: test_src/c/first1000primes/001/primes.c
#include <stdio.h>

int is_prime(int n) {
    if (n < 2) return 0;
    if (n == 2) return 1;
    if (n % 2 == 0) return 0;
    for (int i = 3; i * i <= n; i += 2) {
        if (n % i == 0) return 0;
    }
    return 1;
}

int main() {
    int count = 0;
    int num = 2;
    int last_prime = 0;
    
    while (count < 1000) {
        if (is_prime(num)) {
            last_prime = num;
            count++;
        }
        num++;
    }
    
    printf("%d\n", last_prime);
    return 0;
}

File: test_src/pascal/first1000primes/001/primes.pas
program Primes;

function IsPrime(n: Integer): Boolean;
var
    i: Integer;
begin
    if n < 2 then
        Exit(False);
    if n = 2 then
        Exit(True);
    if (n mod 2) = 0 then
        Exit(False);
    
    i := 3;
    while i * i <= n do
    begin
        if (n mod i) = 0 then
            Exit(False);
        i := i + 2;
    end;
    
    Exit(True);
end;

var
    count, num, lastPrime: Integer;
begin
    count := 0;
    num := 2;
    lastPrime := 0;
    
    while count < 1000 do
    begin
        if IsPrime(num) then
        begin
            lastPrime := num;
            count := count + 1;
        end;
        num := num + 1;
    end;
    
    writeln(lastPrime);
end.
